pallets/storage-provider/src/lib.rs

```diff
@@ -714,6 +720,52 @@ pub mod pallet {
 
             Self::validate_deadline(&current_deadline, &windowed_post)?;
 
+            // record sector as proven
+            let all_sectors = sp.sectors.clone();
+            let deadlines = sp.get_deadlines_mut();
+            deadlines
+                .record_proven(
+                    windowed_post.deadline as usize,
+                    &all_sectors,
+                    windowed_post.partitions.clone(),
+                )
+                .map_err(|e| Error::<T>::GeneralPalletError(e))?;
+
+            // TODO(@th7nder,#592, 19/11/2024): handle faulty and recovered sectors, we don't take them into account now
+            let deadlines = &sp.deadlines;
+            let mut replicas = BoundedBTreeMap::new();
+            // Take all the sectors that were assigned to all of the partitions
+            for partition in &windowed_post.partitions {
+                // Deadline is validated by `Self::validate_deadline`, so we're sure it can be used as an index.
+                let deadline = &deadlines.due[windowed_post.deadline as usize];
+                let sectors = &deadline
+                    .partitions
+                    .get(&partition)
+                    .ok_or(Error::<T>::InvalidPartition)?
+                    .sectors;
+                for sector_number in sectors {
+                    // Sectors stored in the Storage Provider struct should be consistently stored, without breaking invariants.
+                    let sector_info = &sp.sectors[sector_number];
+                    let comm_r = Commitment::<CommR>::from_cid_bytes(&sector_info.sealed_cid)
+                        .expect("CommR to be validated on pre-commit");
+                    let _ = replicas
+                        .try_insert(
+                            *sector_number,
+                            PublicReplicaInfo {
+                                comm_r: comm_r.raw(),
+                            },
+                        )
+                        .map_err(|_| Error::<T>::TooManyReplicas);
```

Comment:
I am a bit confused by this. Why are you capturing the result in `_` instead of unwrapping with `?`?

Reply:
`try_insert()` returns the old value if it was there, I don't mind the old value, so I'm discarding it.

Reply:
You are not unwrapping tho, just mapping `Result<T, E>` to `Result<T, F>` and discarding the `Result<T, F>`. Why not return an error like this:
```suggestion
                    replicas
                        .try_insert(
                            *sector_number,
                            PublicReplicaInfo {
                                comm_r: comm_r.raw(),
                            },
                        )
                        .map_err(|_| Error::<T>::TooManyReplicas)?;
```

Reply:
You're right! Thanks for spotting this one, good catch. I need to add `?`.